subsys/sal/sid_pal/src/sid_mfg_hex_v8.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <sid_mfg_hex_parsers.h>
+#include <sid_hal_memory_ifc.h>
+#include <stdint.h>
+#include <tlv/tlv.h>
+#include <tlv/tlv_storage_impl.h>
+#include <zephyr/sys/byteorder.h>
+#include <string.h>
+#ifdef CONFIG_SIDEWALK_CRYPTO_PSA_KEY_STORAGE
+#include <sid_pal_mfg_store_ifc.h>
+#include <sid_crypto_keys.h>
+#endif
+
+#define MFG_STORE_TLV_TAG_EMPTY 0xFFFF
+
+LOG_MODULE_REGISTER(sid_mfg_parser_v8, CONFIG_SIDEWALK_LOG_LEVEL);
+
+int parse_mfg_raw_tlv(tlv_ctx *tlv, struct mfg_header *mfg_header)
+{
+	if (tlv->end_offset <= tlv->start_offset) {
+		return -EINVAL;
+	}
+
+	uint32_t size = tlv->end_offset - tlv->start_offset;
+
+	int err = tlv_read_start_marker(tlv, (uint8_t *)mfg_header, sizeof(struct mfg_header));
+	if (err != 0) {
+		LOG_ERR("Failed to read marker before MFG data");
+		return -EINVAL;
+	}
+	uint8_t *ram_tlv_data = sid_hal_malloc(size);
+	if (ram_tlv_data == NULL) {
+		return -ENOMEM;
+	}
+	memset(ram_tlv_data, 0xff, size);
+
+	tlv_ctx ram_tlv = { .start_offset = 0,
+			    .end_offset = size,
+			    .tlv_storage_start_marker_size = tlv->tlv_storage_start_marker_size,
+			    .storage = { .ctx = ram_tlv_data,
+					 .read = tlv_storage_ram_read,
+					 .write = tlv_storage_ram_write } };
+
+	uint32_t offset = tlv->start_offset + tlv->tlv_storage_start_marker_size;
+	for (; offset < tlv->end_offset;) {
+		uint8_t key[2];
+		uint8_t size[2];
+		int ret = tlv->storage.read(tlv->storage.ctx, offset, key, 2);
+		if (ret != 0) {
+			LOG_ERR("Failed to read data");
+			sid_hal_free(ram_tlv_data);
+			return -EIO;
+		}
+		offset += 2;
+		uint16_t key_decoded = (key[0] << 8) + key[1];
+		if (key_decoded == MFG_STORE_TLV_TAG_EMPTY) {
+			break;
+		}
+		ret = tlv->storage.read(tlv->storage.ctx, offset, size, 2);
+		if (ret != 0) {
+			LOG_ERR("Failed to read data");
+			sid_hal_free(ram_tlv_data);
+			return -EIO;
+		}
+		offset += 2;
+		uint16_t size_decoded = (size[0] << 8) + size[1];
```

Comment:
minor: don't mix bitwise and arithmetic operator. Repleace with e.g. `(size[0] << 8) | size[1]`